prince_cr/solvers/propagation.py

```diff
@@ -448,29 +452,34 @@ def eqn_deriv_mkl(self, z, state, *args):
             self.current_z_rates = z
 
         res = np.zeros(state.shape)
-        
         if matrix_input:
             state = np.ascontiguousarray(state)
             res = csrmm(1., self.jacobian, state, 0., res)
         else:
             res = csrmv(1., self.jacobian, state, 0., res)
-            
 
         if self.enable_injection_jacobian:
-            res += self.injection(1., z)[:, np.newaxis]
+            try:
+                res += self.injection(1., z)[:, np.newaxis]
+            except ValueError:
+                res += self.injection(1., z)
         
         if self.enable_partial_diff_jacobian:
             conloss = np.zeros_like(self.adia_loss_rates_grid.energy_vector)
             if self.enable_adiabatic_losses:
                 conloss += self.adia_loss_rates_grid.loss_vector(z)
             if self.enable_pairprod_losses:
                 conloss += self.pair_loss_rates_grid.loss_vector(z)
+            try:
+                loss_state = conloss[:, np.newaxis] * state
+            except ValueError:
+                loss_state = conloss * state
```

Comment:
Same as last comment, maybe rather fix in Euler solver
```suggestion
            # catch error due to different shape of res in UHECRPropagationSolverEULER
            try:
                loss_state = conloss[:, np.newaxis] * state
            except ValueError:
                loss_state = conloss * state
```